### PR TITLE
Lima: Darwin: Don't always use en0 for bridged network.

### LINKED ISSUE #1125: Pick a working network interface for the bridged `rd0` vmnet interface
The host interface is specified in `~/Library/Application\ Support/rancher-desktop/lima/_config/networks.yaml`:

```yaml
networks:
  bridged:
    mode: bridged
    interface: en0
```

This must be set before the sudoers file is created:

```console
$ grep en0 /etc/sudoers.d/rancher-desktop-lima
    /opt/rancher-desktop/bin/vde_vmnet --pidfile=/private/var/run/rancher-desktop-lima/bridged_vmnet.pid --vde-group=staff --vmnet-mode=bridged --vmnet-interface=en0 /private/var/run/rancher-desktop-lima/bridged.ctl, \
```

So the default configuration only works if `en0` is connected to a network that provides a DHCP server.

Some suggestions to find the network automatically:

### Using `networksetup`

This will work on older macOS versions like Mojave, but requires parsing plain text output:

```console
$ networksetup -listnetworkserviceorder
An asterisk (*) denotes that a network service is disabled.
(1) Ethernet
(Hardware Port: Ethernet, Device: en0)

(2) Wi-Fi
(Hardware Port: Wi-Fi, Device: en1)

(3) Bluetooth PAN
(Hardware Port: Bluetooth PAN, Device: en4)
[...]
```

This provides all network services in priority order, **and** the device name.

Then query each device in turn:

```console
$ networksetup -getinfo "Ethernet"
DHCP Configuration
Client ID:
IPv6: Automatic
IPv6 IP address: none
IPv6 Router: none
Ethernet Address: 38:c9:86:21:6d:c1
$ networksetup -getinfo "Wi-Fi"
DHCP Configuration
IP address: 192.168.17.21
Subnet mask: 255.255.0.0
Router: 192.168.17.1
Client ID:
IPv6: Automatic
IPv6 IP address: none
IPv6 Router: none
Wi-Fi ID: 28:f0:76:12:69:14
```

This shows that `en0` has no IP address, and that `en1` should be configured for the bridged network.

### Using `system_profiler SPNetworkDataType -json`

The `-json` option is only supported on macOS Catalina and later. It is possible to parse the regular output, but feels rather brittle.

I don't see this documented, but it looks like the network devices are also returned in priority order:

```console
$ system_profiler SPNetworkDataType -json | jq '.SPNetworkDataType[] | [._name, .interface, .ip_address]'
[
  "Ethernet",
  "en0",
  null
]
[
  "Wi-Fi",
  "en1",
  [
    "192.168.17.21"
  ]
]
[
  "Bluetooth PAN",
  "en4",
  null
]
...
```

This means we could select the device to use like this:

```console
$ system_profiler SPNetworkDataType -json | jq -r 'first(.SPNetworkDataType[] | select(.ip_address) | .interface)'
en1
```

### Roaming

This is out-of-scope for this issue, but I wanted to mention that we'll eventually have to deal with roaming too: A laptop can travel to different networks, so the bridged network address may change. Ethernet may be plugged in and Wi-Fi disconnected. Rancher Desktop should detect these situations and prompt the user to restart the VM to adapt.

### PR DESCRIPTION
**Pull request by mook-as**
On some systems, en0 is not connected; in that case, we need to find the correct network to bridge onto.  This implements the detection.

Fixes #1125.

### CODE REVIEW

File: src/k8s-engine/lima.ts [hunk 1]
```diff
@@ -476,12 +507,19 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
     });
 
     if (os.platform() === 'darwin') {
-      config.networks = [
-        {
-          lima:      'bridged',
-          interface: 'rd0',
-        },
-      ];
+      const hostNetwork = (await this.getDarwinHostNetworks()).find(n => n.IPv4 && n.dhcp);
```
Comment (jandubois) [inline src/k8s-engine/lima.ts:510]: Not sure if testing for `n.IPv4` is useful; that field seems always there, even if it just has a `ConfigMethod` field.

Maybe we should test for `n.IPv4.Addresses` to make sure they got at least an address? And or `AdditionalRoutes`?

Also, will `find` return just a single entry, even when there are multiple matches?

File: src/k8s-engine/lima.ts [hunk 2]
```diff
@@ -917,20 +955,48 @@ ${ commands.join('\n') }
...
-      const runFile = data?.paths?.varRun ?? '';
+    let config: LimaNetworkConfiguration;
 
-      if (runFile.includes('/rancher-desktop')) {
-        // Assume if there's a paths.varRun setting mentioning "rancher-desktop" there's no need to replace it.
-        return;
+    try {
+      config = yaml.parse(await fs.promises.readFile(networkPath, 'utf8'));
+      if (config?.paths?.varRun !== NETWORKS_CONFIG.paths.varRun) {
+        console.log('Lima network configuration has unexpected contents; it will be replaced.');
```
Comment (jandubois) [inline src/k8s-engine/lima.ts:963]: Maybe add the old contents to the log, so you have a (brief) chance to recover it?

File: src/k8s-engine/lima.ts (hunk 1 above)
Comment (mook-as) [reply src/k8s-engine/lima.ts:510]: Yes, [`find`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) returns the first result, or `undefined`; [`filter`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) returns an array of (possibly multiple, possibly zero) matches.

**Comment by jandubois** [comment]
Testing the PR shows an additional undefined entry in `networks.yaml`:

```
  33   │   bridged_undefined:
  34   │     mode: bridged
  35   │     interface: null
```

**Review by jandubois — APPROVED** [review]
LGTM and works as expected

**Pull request merged**